python/ray/train/torch/torch_trainer.py

```diff
@@ -93,6 +93,40 @@ def train_loop_per_worker():
     To save a model to use for the ``TorchPredictor``, you must save it under the
     "model" kwarg in ``Checkpoint`` passed to ``session.report()``.
 
+    .. note::
+        If you are wrapping your ``model`` with ``prepare_model``, save ``model`` to
+        session is equivalent to saving ``model.module`` to session.
+        When you load from a saved checkpoint, make sure that you first
+        load ``state_dict`` to model before calling ``prepare_model``.
+        Otherwise, you will run into errors like
+        ``Error(s) in loading state_dict for DistributedDataParallel:
+        Missing key(s) in state_dict: "module.conv1.weight", ...``. See snippet below.
+
+        .. testcode::
+
+            from torchvision.models import resnet18
+            from ray.air import session
+            from ray.air.checkpoint import Checkpoint
+            import ray.train as train
+
+            def train_func():
+                ...
+                model = resnet18()
+                # The following wraps model with DDP.
+                # An effect of that is in `state_dict`, keys are prefixed by
+                # `module.`. For example:
+                # `layer1.0.bn1.bias` becomes `module.layer1.0.bn1.bias`.
```

Comment:
```suggestion
                # Prepare model for distributed training 
                # (e.g. wrap in DistributedDataParallel)
```

the exaplanation with `module.layer` should IMO be in the docstring, not in this inline comment